File: windows/src/GoalMaker.Core/Sync/ColumnKind.cs
namespace GoalMaker.Core.Sync;

/// <summary>How a synced column is stored locally and sent on the wire (contracts/schemas/synced-tables.json).</summary>
public enum ColumnKind
{
    Text,
    Integer,
    Real,
    Boolean,
    Timestamp,
    Date,
    Time,

    /// <summary>A JSON value: its JSON text in the replica, the value itself on the wire.</summary>
    Json,
}

File: android/app/src/main/kotlin/com/goalmaker/app/domain/sync/ColumnKind.kt
package com.goalmaker.app.domain.sync

import java.util.Locale

/** How a synced column is stored in the replica and sent on the wire (contracts/schemas/synced-tables.json). */
enum class ColumnKind {
    TEXT,
    INTEGER,
    REAL,
    BOOLEAN,
    TIMESTAMP,
    DATE,
    TIME,
    ;

    companion object {
        fun parse(value: String): ColumnKind = valueOf(value.uppercase(Locale.ROOT))
    }
}
